### ISSUE TITLE
Missing Logging of Exceptions in IHostedService without await

### ISSUE THREAD

**Issue by Tripletri** [dotnet/runtime#84559]
### Description

When an exception occures in IHostedService without calling `await`, it is not logged to the `ILogger`.
This make it difficult to identify the reason for the application shutdown when writing logs to an external service.

### Reproduction Steps

```cs
IHost host = Host.CreateDefaultBuilder(args)
    .ConfigureServices(services =>
    {
        services.AddHostedService<Worker>();
    })
    .Build();

await host.RunAsync();

public class Worker : IHostedService
{
    public Task StartAsync(CancellationToken cancellationToken) =>
        throw new Exception();

    public Task StopAsync(CancellationToken cancellationToken) =>
        Task.CompletedTask;
}
```

### Expected behavior

When an exception occurs within the `StartAsync` of `IHostedService`, whether or not method has `await`, the exception should be logged to the `ILogger`

### Actual behavior

When an exception occurs in the `StartAsync` without calling `await`, the exception is not logged to the `ILogger`

### Configuration

Sdk: Microsoft.NET.Sdk.Worker
Framework: .NET 7.0.3

### Other information

It is also lead to misunderstanding of `BackgroundServiceExceptionBehavior.Ignore`, which continue to stop host when exception occurs.

This can be relative to #67146

**Comment by buyaa-n** [comment]
It looks we did not had any exception hanlding for running hosted services:
https://github.com/dotnet/runtime/blob/417a01523fb75fa2688b6407db2e71ded7063f72/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L68-L77

This is updated recently with https://github.com/dotnet/runtime/pull/84048 and https://github.com/dotnet/runtime/pull/85191:
https://github.com/dotnet/runtime/blob/71c9c73a776b692b517ea7b6c42483d09cd41531/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L69-L123

So, the issues you are seeing might have already fixed. The repro you provided doesn't have a logger and the await that causing the logging work, so I cannot check. @Tripletri please tryout your code with the [.NET 8 preview 5 bits](https://github.com/dotnet/installer#installers-and-binaries) and let us know.

**Comment by Tripletri** [comment]
Yes, the issue is fixed in `8.0.100-preview.5.23301.12`. Thanks!
